pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -1096,19 +1096,10 @@ impl LayerInner {
 
         match rx.await {
             Ok(Ok(res)) => Ok(res),
-            Ok(Err(e)) => {
-                // sleep already happened in the spawned task, if it was not cancelled
-                match e.downcast_ref::<remote_storage::DownloadError>() {
-                    // If the download failed due to its cancellation token,
-                    // propagate the cancellation error upstream.
-                    Some(remote_storage::DownloadError::Cancelled) => {
-                        Err(DownloadError::DownloadCancelled)
-                    }
-                    // FIXME: this is not embedding the error because historically it would had
-                    // been output to compute, however that is no longer the case.
-                    _ => Err(DownloadError::DownloadFailed),
-                }
-            }
+            Ok(Err(e)) => match e {
```

Comment:
The logging in question happens on L1087R1087. Maybe just change it to not log with stacktrace if it's a cancellation error?

Reply:
https://github.com/neondatabase/neon/pull/8224#issuecomment-2202692465

Reply:
Aah yes, because I had not done the fixme I asked you to keep in the other thread :)